llvm/lib/Target/AMDGPU/AMDGPUCombine.td

```diff
@@ -15,8 +15,9 @@ def fmin_fmax_legacy_matchdata : GIDefMatchData<"FMinFMaxLegacyInfo">;
 let Predicates = [HasFminFmaxLegacy] in
 def fcmp_select_to_fmin_fmax_legacy : GICombineRule<
   (defs root:$select, fmin_fmax_legacy_matchdata:$matchinfo),
```

Comment:
Am I supposed to list `$fcmp` in the defs since I am passing it into my match function? Or doesn't that count as an "external" use of `$fcmp`?

Reply:
No. I prefer to only list the `$root`. defs is probably definitions and not def in SSA.